leaderelection/src/main/java/io/scalecube/leaderelection/RaftStateMachine.java

```diff
@@ -0,0 +1,58 @@
+package io.scalecube.leaderelection;
+
+
+import unquietcode.tools.esm.EnumStateMachine;
+
+/**
+ * RAFT state machine dealing with state transition FOLLOWER, CANNIDATE, LEADER.
+ * 
+ * @author Ronen Nachmias
+ */
+public abstract class RaftStateMachine {
```

Comment:
design decesion to remove all the noise from leader election class that handles the actual logic and not the boilerplate

or at least less boilerplate ...